cloud-sql/mysql/servlet/src/main/java/com/example/cloudsql/IndexServlet.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+
+package com.example.cloudsql;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+@WebServlet(name = "Index", value = "")
+public class IndexServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws IOException, ServletException {
+    // Extract the pool from the Servlet Context, reusing the one that was created
+    // in the ContextListener when the application was started
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    int tabCt;
+    int voteCt;
+    List<Vote> recentVotes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      // PreparedStatements are compiled by the database immediately and executed at a later date.
+      // Most databases cache previously compiled queries, which improves efficiency.
+      PreparedStatement voteStmt =  conn.prepareStatement(
+          "SELECT canidate, time_cast FROM votes ORDER BY time_cast DESC LIMIT 5");
+      // Execute the statement
+      ResultSet voteResults = voteStmt.executeQuery();
+      // Convert a ResultSet into Vote objects
+      while (voteResults.next()) {
+        String candidate = voteResults.getString(1);
+        Timestamp timeCast = voteResults.getTimestamp(2);
+        recentVotes.add(new Vote(candidate, timeCast));
+      }
+
+      // PreparedStatements can also be executed multiple times with different arguments. This can
+      // improve efficiency, and project a query from being vulnerable to an SQL injection.
+      PreparedStatement voteCtStmt = conn.prepareStatement(
+          "SELECT COUNT(vote_id) FROM votes WHERE canidate=?");
+
+      voteCtStmt.setString(1, "tabs");
+      ResultSet tabResult = voteCtStmt.executeQuery();
+      tabResult.next(); // Move to the first result
+      tabCt = tabResult.getInt(1);
+
+      voteCtStmt.setString(1, "spaces");
+      ResultSet spacesResult = voteCtStmt.executeQuery();
+      spacesResult.next(); // Move to the first result
+      voteCt = spacesResult.getInt(1);
+
+    } catch (SQLException e) {
+      // If something goes wrong, the application needs to react appropriately. This might mean
+      // getting a new connection and executing the query again, or it might mean redirecting the
+      // user to a different page to let them know something went wrong.
+      throw new ServletException("Unable to successfully connect to the database. Please check the "
```

Comment:
pls propagate `e` in `ServletException`

See https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/ServletException.html#ServletException(java.lang.String,%20java.lang.Throwable)

Reply:
IRL you'd probably log the event, maybe add to some monitor value as well before propagating.